persist/sqlite/consensus.go

```diff
@@ -1204,3 +1203,80 @@ func (s *Store) UpdateChainState(reverted []chain.RevertUpdate, applied []chain.
 		return nil
 	})
 }
+
+// ResetChainState implements explorer.Store
+func (s *Store) ResetChainState() error {
+	if err := s.transaction(func(tx *txn) error {
+		if _, err := tx.Exec(`PRAGMA defer_foreign_keys=ON`); err != nil {
```

Comment:
don't forget to turn it off

Reply:
It's a `defer` pragma now so it will not affect the foreign key checks beyond this transaction